point-of-sale-app/inventory/src/main/java/com/google/abmedge/inventory/InventoryController.java

```diff
@@ -50,13 +94,92 @@ public String readiness() {
+    for (PurchaseItem pi : purchaseList) {
+      UUID itemId = pi.getItemId();
+      Optional<Item> loadedItem = activeConnector.getById(itemId);
+      if (loadedItem.isPresent()) {
+        Item i = loadedItem.get();
+        long currentQuantity = i.getQuantity();
+        if (currentQuantity >= pi.getItemCount()) {
+          long newQuantity = currentQuantity - pi.getItemCount();
+          i.setQuantity(newQuantity);
+          updated &= activeConnector.update(i);
+          LOGGER.info(String.format("Updated item '%s - %s' with new quantity '%s'",
+              itemId, i.getName(), newQuantity));
+          continue;
+        }
+        LOGGER.error(String.format(
+            "Failed to update item '%s - %s'. "
+                + "The requested count '%s' is more than whats available '%s'",
+            itemId, i.getName(), pi.getItemCount(), currentQuantity
+        ));
+      } else {
+        LOGGER.warn(String.format("Update attempt with invalid item id: '%s'", itemId));
+        updated = false;
+      }
+    }
+    if (!updated) {
+      LOGGER.error("Failed to update one or more items in the purchase list!");
+      return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+    return new ResponseEntity<>(HttpStatus.OK);
+  }
+
+  private void initInventoryItems() {
+    String inventoryList = System.getenv(INVENTORY_ITEMS_ENV_VAR);
+    if (StringUtils.isBlank(inventoryList)) {
+      LOGGER.warn(String.format(
+          "No items found under inventory list env var '%s'", INVENTORY_ITEMS_ENV_VAR));
+      return;
+    }
+    LOGGER.debug(inventoryList);
+    Yaml yaml = new Yaml(new Constructor(Inventory.class));
+    Inventory inventory = yaml.load(inventoryList);
+    inventory.getItems().forEach(i -> {
+      i.setId(UUID.randomUUID());
+      activeConnector.insert(i);
+      LOGGER.info(String.format("Inserting new item: %s", i.toString()));
+    });
+  }
+
+  private void initConnectorType() {
+    String connectorType = System.getenv(CONNECTOR_TYPE_ENV_VAR);
+    if (StringUtils.isBlank(connectorType) || !inventoryMap.containsKey(connectorType)) {
+      LOGGER.warn(String.format("'%s' environment variable is not set; "
+          + "thus defaulting to: %s", CONNECTOR_TYPE_ENV_VAR, IN_MEMORY_CONNECTOR));
+      connectorType = IN_MEMORY_CONNECTOR;
+    }
+    activeConnector = inventoryMap.get(connectorType);
+    LOGGER.info(String.format("Active connector type is: %s", connectorType));
   }
 
-  @RequestMapping("/all")
-  public String all() {
-    return "Everything is new";
+  private void initItemsType() {
+    activeItemsType = System.getenv(ACTIVE_TYPE_ENV_VAR);
+    if (StringUtils.isBlank(activeItemsType)) {
+      LOGGER.warn(String.format("'%s' environment variable is not set; "
+          + "thus defaulting to: %s", ACTIVE_TYPE_ENV_VAR, ALL_ITEMS));
+      activeItemsType = ALL_ITEMS;
+    }
+    LOGGER.info(String.format("Active items type is: %s", activeItemsType));
```

Comment:
what are "active" items?

Reply:
Added a one line comment in https://github.com/GoogleCloudPlatform/anthos-edge-usecases/pull/2/commits/03fc0b4ed6c19565fe0a599a743e709052fbc1be but this is also explained in the comment added in https://github.com/GoogleCloudPlatform/anthos-edge-usecases/commit/945912a1e0b704603502a37fb783e622f62d6454